### PR TITLE
Increase coverage

### PR DESCRIPTION
**Pull request by markstory**
These changes should increase code coverage. I also found an issue in UuidType around array to string conversion errors.

I've also attempted to fix the strange build errors in PHP 5.4 and 5.5

### CODE REVIEW

**Comment by jeremyharris** [comment]
:+1: nice work Mark

**Comment by markstory** [comment]
Not sure why coveralls failed. Coverage is ~= 90% now.

File: .travis.yml [hunk 1]
```diff
@@ -64,6 +64,8 @@ before_script:
   - sh -c "if [ '$COVERALLS' = '1' ]; then mkdir -p build/logs; fi"
 
   - sh -c "if [ '$HHVM' != '1' ]; then echo 'extension = memcached.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi"
+  - sh -c "if [ '$HHVM' != '1' ]; then echo 'extension = redis.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi"
+  - sh -c "if [ '$HHVM' != '1' ]; then echo 'extension = apc.so' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi"
```
Comment (antograssiot) [inline .travis.yml]: this is probably only needed for php 5.4 if ever ?

File: tests/TestCase/Database/Type/IntegerTypeTest.php [hunk 2]
```diff
@@ -69,6 +69,8 @@ public function testToPHP()
      */
     public function testToDatabase()
     {
+        $this->asserNull($this->type->toDatabase(null, $this->driver));
```
Comment (antograssiot) [inline tests/TestCase/Database/Type/IntegerTypeTest.php]: typo asser**t**Null

File: tests/TestCase/Database/Type/StringTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        $this->assertSame('toString called', $this->type->toDatabase($obj, $this->driver));
+    }
+
+    /**
+     * Tests that passing an invalid value will throw an exception
+     *
+     * @expectedException InvalidArgumentException
+     * @return void
+     */
+    public function testToDatabseInvalidArray()
```
Comment (antograssiot) [inline tests/TestCase/Database/Type/StringTypeTest.php]: not important but typo in Databse -> Database and in BoolTypeTest.php also

**Comment by lorenzo** [comment]
Great job!

**Pull request merged**